cloudformation/template.yml

```diff
@@ -336,6 +378,13 @@ Resources:
         SecurityGroupIds:
           - !Ref ContainerInstanceSecurityGroup
         Subnets: !Ref SubnetIds
+        LaunchTemplate:
+          !If
+            - UseGpuLaunchTemplate
+            -
+              LaunchTemplateId: !Ref GpuLaunchTemplate
+              Version: !Ref GpuLaunchTemplateVersion
+            - !Ref "AWS::NoValue"
```

Comment:
I'd never seen this syntax before, but `!If` is basically just a ternary operator: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-conditions.html#intrinsic-function-reference-conditions-if

Reply:
Neither have I. This looks like exactly what we're looking for in Terraform land with regard to ASG `desired_capacity`.